### PR TITLE
[DEPRECATED] Support IRCv3.2 metadata extension

### PR DESCRIPTION
**Pull request by mbax**
http://ircv3.net/specs/core/metadata-3.2.html

**DONE**
- Command
- Restrict potential keys in command
- Detect support in ISUPPORT

**TODO**
- `metadata-notify`
- Numeric events

### CODE REVIEW

**Comment by mbax** [comment]
Suggestions:
unSet -> unset
Better javadocs about start

File: src/main/java/org/kitteh/irc/client/library/command/MetadataCommand.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        @Override
+        public void execute() {
+            this.getClient().sendRawLine(this.getStart() + "LIST");
+        }
+    }
+
+    /**
+     * Sets the metadata for a given target.
+     */
+    public class Set extends MetadataCommand {
```
Comment (kashike) [inline src/main/java/org/kitteh/irc/client/library/command/MetadataCommand.java]: Should this not be `static`?

File: src/main/java/org/kitteh/irc/client/library/command/MetadataCommand.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+                builder.append(key);
+            }
+            this.getClient().sendRawLine(builder.toString());
+        }
+    }
+
+    /**
+     * Lists the metadata for a given target.
+     */
+    public class List extends MetadataCommand {
```
Comment (kashike) [inline src/main/java/org/kitteh/irc/client/library/command/MetadataCommand.java]: Should this not be `static`?

File: src/main/java/org/kitteh/irc/client/library/command/MetadataCommand.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        @Override
+        public void execute() {
+            this.getClient().sendRawLine(this.getStart() + "CLEAR");
+        }
+    }
+
+    /**
+     * Gets the requested metadata for a given target.
+     */
+    public class Get extends MetadataCommand {
```
Comment (kashike) [inline src/main/java/org/kitteh/irc/client/library/command/MetadataCommand.java]: Should this not be `static`?

File: src/main/java/org/kitteh/irc/client/library/command/MetadataCommand.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+import java.util.regex.Pattern;
+
+/**
+ * How meta.
+ */
+public abstract class MetadataCommand extends Command {
+    /**
+     * Clears all metadata for a given target.
+     */
+    public class Clear extends MetadataCommand {
```
Comment (kashike) [inline src/main/java/org/kitteh/irc/client/library/command/MetadataCommand.java]: Should this not be `static`?

File: src/main/java/org/kitteh/irc/client/library/implementation/ManagerISupport.java [hunk 5]
```diff
@@ -220,6 +220,28 @@ private ISupportChanTypes(@Nonnull Client client, @Nonnull String name, @Nonnull
         }
     }
 
+    private static final class ISupportMetadata extends IRCISupportParameter implements ISupportParameter.Metadata {
+        private Optional<Integer> limit;
```
Comment (kashike) [inline src/main/java/org/kitteh/irc/client/library/implementation/ManagerISupport.java]: `final`?

Comment (mbax) [inline src/main/java/org/kitteh/irc/client/library/implementation/ManagerISupport.java]: seems reasonable

**Comment by mbax** [comment]
Closing until the day comes it's no longer deprecated and this PR is still helpful to the new spec.

**Pull request closed without merging**